ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -607,9 +606,7 @@ private Long incrementalDump(Path dumpRoot, DumpMetaData dmd, Path cmRoot, Hive
     dmd.setDump(DumpType.INCREMENTAL, work.eventFrom, lastReplId, cmRoot, executionId);
     // If repl policy is changed (oldReplScope is set), then pass the current replication policy,
     // so that REPL LOAD would drop the tables which are not included in current policy.
-    if (work.oldReplScope != null) {
-      dmd.setReplScope(work.replScope);
-    }
+    dmd.setReplScope(work.replScope);
```

Comment:
Earlier current ReplicationScope was being passed for load to consume and figure out if  it need to drop extra tables. For which it makes a call to metastore to get all  tables and filter the names by applying the expression in ReplicationScope. With this change, this comparison will happen all the times.  While this will still work, it will introduce a new cost to get all tables and compare against the  load. Can we maintain two  expressions or one expression and one flag to convey if there have  been a change in the  expression or not?